contracts/actions/liquity/LiquitySupply.sol

```diff
@@ -0,0 +1,65 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity =0.7.6;
+pragma experimental ABIEncoderV2;
+
+import "../../interfaces/liquity/IBorrowerOperations.sol";
+import "../ActionBase.sol";
+
+contract LiquitySupply is ActionBase {
+
+    address constant BorrowerOperationsAddr = 0x24179CD81c9e782A4096035f7eC97fB8B783e007;
+
+    /// @inheritdoc ActionBase
+    function executeAction(
+        bytes[] memory _callData,
+        bytes[] memory _subData,
+        uint8[] memory _paramMapping,
+        bytes32[] memory _returnValues
+    ) public payable virtual override returns (bytes32) {
+        (uint256 _amount, address _upperHint, address _lowerHint) = parseInputs(_callData);
+
+        _amount = _parseParamUint(_amount, _paramMapping[0], _subData, _returnValues);
+
+        _amount = _liquitySupply(_amount, _upperHint, _lowerHint);
+        return bytes32(_amount);
+    }
+
+    /// @inheritdoc ActionBase
+    function executeActionDirect(bytes[] memory _callData) public virtual payable override {
+        (uint256 _amount, address _upperHint, address _lowerHint) = parseInputs(_callData);
+
+        _liquitySupply(_amount, _upperHint, _lowerHint);
+    }
+
+    /// @inheritdoc ActionBase
+    function actionType() public pure virtual override returns (uint8) {
+        return uint8(ActionType.STANDARD_ACTION);
+    }
+
+    //////////////////////////// ACTION LOGIC ////////////////////////////
+
+    /// @notice Send ETH as collateral to a trove
+    function _liquitySupply(uint256 _amount, address _upperHint, address _lowerHint) internal returns (uint256) {
+        IBorrowerOperations(BorrowerOperationsAddr).addColl{value: _amount}(_upperHint, _lowerHint);
```

Comment:
All DFS actions work only with tokens, no direct ETH sends. If the protocol accepts only ETH, unwrap weth and put int eth, but always assume actions work with weth